ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/IntegrityChecker.scala

```diff
@@ -128,24 +144,37 @@ class IntegrityChecker[LogResult](commitStrategySupport: CommitStrategySupport[L
     newLoggingContext { implicit loggingContext =>
       val feedHandleResourceOwner = for {
         indexerFactory <- ResourceOwner
-          .forFuture(() =>
-            migrateAndStartIndexer(config, readService, metrics, LfValueTranslation.Cache.none))
+          .forFuture(
+            () =>
+              migrateAndStartIndexer(
+                indexerConfig,
+                readService,
+                metrics,
+                LfValueTranslation.Cache.none,
+            )
+          )
         indexer <- indexerFactory
         feedHandle <- indexer.subscription(readService)
-      } yield feedHandle
+      } yield (feedHandle, System.nanoTime())
 
       // Wait for the indexer to finish consuming the state updates.
       // This works because ReplayingReadService.stateUpdates() closes the update stream
       // when it is done streaming the recorded updates, and IndexFeedHandle.complete()
       // completes when it finishes consuming the state update stream.
       // Any failure (e.g., during the decoding of the recorded state updates, or
       // during the indexing of a state update) will result in a failed Future.
-      feedHandleResourceOwner.use(_.completed())
+      feedHandleResourceOwner.use {
+        case (feedHandle, startTime) => Future.successful(startTime).zip(feedHandle.completed())
+      }
     }.transform {
-      case Success(value) =>
-        println(s"Successfully indexed all updates.".green)
-        println()
-        Success(value)
+      case Success((startTime, _)) =>
+        Success {
+          println(s"Successfully indexed all updates.".green)
+          val durationSeconds = Duration.fromNanos(System.nanoTime() - startTime).toSeconds
+          println(
```

Comment:
To improve readability please move the calculation of updates/second into a `val`.

Reply:
Made a small change to fix this hard-to-read multi-line here.